docs/cloud-init.md

```diff
@@ -65,6 +65,67 @@ spec:
         os: hvm
     type: qemu
 ```
+
+### NoCloud with UserData stored in k8s secret
+
+Instead of placing the userdata directly into the VM definition, another option
+is to reference the userdata in a k8s secret object.
+
+To use this method, the user must create an Opaque type kubernetes secret with
+a key named 'userdata' containing a value representing the base64 encoded
+userdata. The name of the kubernetes secret is then referenced in the CloudInit
+object on the VM spec.
```

Comment:
@rmohr  per our out of band discussion, I added a segment in the docs about how the k8s secret is constructed using the 'userdata' key.  The example in this doc shows how to create the secret as well.